packages/client/src/client.js

```diff
@@ -67,6 +68,11 @@ export default class PercyClient {
 
   // Stringifies client and environment info.
   userAgent() {
+    if (!this.envWarning && (this.clientInfo.size === 0 || this.environmentInfo.size === 0)) {
```

Comment:
By moving the initial values and utilizing the existing methods, at this point we should be certain that these Sets only contain truthy and unique values. This way empty strings will no longer work around the warning.